### ISSUE TITLE
kubelite restarting over and over, consuming lots of resources

### ISSUE THREAD

**Issue by mthaddon** [canonical/microk8s#2067]
Kubelite seems to be restarting over and over, and consuming lots of resources (consistently over 100% CPU). I've attached the inspection report.

[inspection-report-20210301_101231.tar.gz](https://github.com/ubuntu/microk8s/files/6060438/inspection-report-20210301_101231.tar.gz)

**Comment by ktsakalozos** [comment]
Hi @mthaddon. I see in the logs that kube-proxy is complaining with:
```
Mar 01 10:12:00 tenaya microk8s.daemon-kubelite[147209]: F0301 10:12:00.434570  147209 server.go:495] unable to create proxier: CIDR 10.1.0.0/16 has incorrect IP version: expect isIPv6=true
```
I suspect the kube-proxy is picking up the IPv6 address of the default interface. 

Does this happen with the kubeproxy of the v1.20 track? `--channel=1.20/stable`

**Comment by mthaddon** [comment]
I don't seem to have this problem with v1.20 from the stable channel.

I ran `sudo snap remove --purge microk8s` and then `sudo snap remove --purge microk8s` fwiw.

**Comment by ktsakalozos** [comment]
Could you share an inspection tarball from MicroK8s of the 1.20 track? I am interested to see what the kube-proxy logs say.

**Comment by mthaddon** [comment]
[inspection-report-20210301_151357.tar.gz](https://github.com/ubuntu/microk8s/files/6062195/inspection-report-20210301_151357.tar.gz)

**Comment by codetheweb** [comment]
I believe I may have run into something similar. I rebooted my node (microk8s wasn't pinned to a specific version before and snap upgraded to 1.21) and everything fell apart; kubelite was pinning the CPU.

Downgraded back to 1.20 and everything seems fine now. I've attached a report in case it helps.

[inspection-report-20210419_223325.tar.gz](https://github.com/ubuntu/microk8s/files/6339443/inspection-report-20210419_223325.tar.gz)
